internal/fs/fs.go

```diff
@@ -642,14 +642,37 @@ func (fs *fileSystem) lookUpOrCreateInodeIfNotStale(ic inode.Core) (in inode.Ino
 		}
 
 		var ok bool
-		in, ok = fs.implicitDirInodes[ic.FullName]
-		// If we don't have an entry, create one.
-		if !ok {
-			in = fs.mintInode(ic)
-			fs.implicitDirInodes[in.Name()] = in.(inode.DirInode)
+		var maxTriesToCreateInode = 3
```

Comment:
Just want to confirm, with this code changes it will handle the case if we perform 3 concurrent operations (2 forget iNode and CreateInode in progress). If we add one more concurrent forget inode operations, it'll crash. Right? This will not go to the dead-lock at least in this workflow.

Reply:
I am returning nil incase max tries are exhausted. So we will end up returning error to kernel and no crash will happen.